ui-admin/src/study/kits/kitcollection/KitScanner.tsx

```diff
@@ -0,0 +1,303 @@
+  const [enrollee, setEnrollee] = useState<Enrollee>()
+  const [enrolleeCodeError, setEnrolleeCodeError] = useState<string>()
+  const [isEnrolleeIdentityConfirmed, setIsEnrolleeIdentityConfirmed] = useState(false)
+  const [kitBarcode, setKitBarcode] = useState<string | undefined>()
+  const [kitCodeError, setKitCodeError] = useState<string>()
+
+  const [enableManualOverride, setEnableManualOverride] = useState(false)
+  const [enrolleeShortcodeOverride, setEnrolleeShortcodeOverride] = useState<string>()
+
+  const { user } = useUser()
+
+  const assignKit = async () => {
+    if (!enrollee || !kitBarcode) { return }
+    doApiLoad(async () => {
+      await Api.createKitRequest(
+        studyEnvContext.portal.shortcode,
+        studyEnvContext.study.shortcode,
+        studyEnvContext.currentEnv.environmentName,
+        enrollee.shortcode,
+        {
+          kitType: 'SALIVA',
+          distributionMethod: 'IN_PERSON',
+          kitBarcode,
+          skipAddressValidation: false
+        }
+      )
+      Store.addNotification(successNotification('Kit successfully assigned'))
+    }, {
+      setIsError: error => {
+        if (error) {
+          setEnrolleeCodeError('Error assigning kit')
+        }
+      }
+    })
+  }
+
+  const collectKit = async () => {
+    if (!enrollee || !kitBarcode) { return }
+    doApiLoad(async () => {
+      await Api.collectKit(
+        studyEnvContext.portal.shortcode,
+        studyEnvContext.study.shortcode,
+        studyEnvContext.currentEnv.environmentName,
+        enrollee.shortcode,
+        {
+          kitBarcode
+        }
+      )
+      Store.addNotification(successNotification('Kit successfully collected'))
+    }, {
+      setIsError: error => {
+        if (error) {
+          setEnrolleeCodeError('Error collecting kit')
+        }
+      }
+    })
+  }
+
+  const loadEnrollee = (enrolleeShortcode: string) => {
+    doApiLoad(async () => {
+      const loadedEnrollee = await Api.getEnrollee(
+        studyEnvContext.portal.shortcode,
+        studyEnvContext.study.shortcode,
+        studyEnvContext.currentEnv.environmentName,
+        enrolleeShortcode
+      )
+      if (loadedEnrollee.consented) {
+        setEnrollee(loadedEnrollee)
+        setIsEnrolleeIdentityConfirmed(false)
+      } else {
+        setEnrolleeCodeError('Enrollee has not consented to the study')
+        setEnrollee(undefined)
+      }
+    }, {
+      setIsError: error => {
+        if (error) {
+          setEnrolleeCodeError('Error loading enrollee. Did you scan the correct QR code?')
+          setEnrollee(undefined)
+        }
+      }
+    })
+  }
+
+  //TODO: JN-1295, while under development, we'll only allow superusers to access this page
+  if (!user?.superuser) {
+    return <div className="m-2">
+      <div className="alert alert-danger" role="alert">
+          You do not have permission to access this page.
+      </div>
+    </div>
+  }
```

Comment:
There's now a `<RequireUserPermission superuser noPermsMessage={<></>}>` component you could use for this